code/game/objects/structures/mirror.dm

```diff
@@ -239,7 +204,29 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/mirror, 28)
 			if(!user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
 				return TRUE
 			if(hairchoice == "Style") //So you just want to use a mirror then?
-				return ..()
+				//handle facial hair (if necessary)
+				if(amazed_human.gender != FEMALE)
+					var/new_style = tgui_input_list(user, "Select a facial hairstyle", "Grooming", GLOB.facial_hairstyles_list)
+					if(isnull(new_style))
+						return TRUE
+					if(!user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
+						return TRUE //no tele-grooming
+					amazed_human.facial_hairstyle = new_style
+				else
+					amazed_human.facial_hairstyle = "Shaved"
+
+				//handle normal hair
+				var/new_style = tgui_input_list(user, "Select a hairstyle", "Grooming", GLOB.hairstyles_list)
+				if(isnull(new_style))
```

Comment:
could you not just do 
```suggestion
				if(!new_style)
```

Reply:
I choose isnull because I like the explicitness